### PR TITLE
fix: [CDS-77892]: Added support to configFile.getAsBase64 encoding files without secrets

### PR DESCRIPTION
**Pull request by ivan-83**
Due to the fact that we are not supporting secrets in files while encoding it by using configFile.getAsBase64 in CG. 
The approach is to encode the whole file if it doesn't contain a secret and pass it to the delegate. 

Reported customer use case. 

Screenshot 2023-09-15 at 19 29 13

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52975)

### CODE REVIEW

**Review by acasian-harness — DISMISSED** [review]

**Comment by ivan-83** [comment]
trigger uts

**Comment by ivan-83** [comment]
trigger gitleaks

**Comment by ivan-83** [comment]
trigger checkstylecodeformat

**Comment by ivan-83** [comment]
trigger pmd

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52975)* status: Some of the checks are too old. 3 file(s) to review.

**Comment by yogesh-chauhan** [comment]
If the file content contains a secret, and also contains `}` it might not work ?

**Comment by ivan-83** [comment]
trigger uts

**Comment by ivan-83** [comment]
trigger commonchecks

**Comment by ivan-83** [comment]
trigger messagecheck

**Review by ivan-83 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52975)* status: 6 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

*[a discussion](https://reviewable.io/reviews/harness/harness-core/52975#-NfG8Vmb5FUU8R8NC-Lq:-NfKsnoT-8Dx-dPYaK6k:b2nign6) (no related file):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

If the file content contains a secret, and also contains `}` it might not work ?
</blockquote></details>

yes

___

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java [hunk 1]
```diff
@@ -719,6 +721,19 @@ private Optional<String> getReleaseNameFromService(Ambiance ambiance) {
 
   public String getFileContentAsBase64(Ambiance ambiance, String scopedFilePath, long allowedBytesFileSize) {
     String content = getFileContentAsString(ambiance, scopedFilePath, allowedBytesFileSize);
+    boolean fileContentContainSecret = containsSecret(content);
```
Comment (acasian-harness) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java:724]: do we have any ut's covering the use cases? Quite tricky if nextime we will need to fix and without good knowledge if we didn't broke any existing flow.

**Comment by ivan-83** [comment]
trigger uts

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java (hunk 1 above)
Comment (ivan-83) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java:728]: Added more asserts to check whether the content contains the secret.

**Comment by ivan-83** [comment]
trigger ut1

**Comment by ivan-83** [comment]
trigger ut8

**Comment by ivan-83** [comment]
trigger commonchecks

**Comment by ivan-83** [comment]
trigger gitleaks

**Comment by ivan-83** [comment]
trigger codebasehash

**Comment by ivan-83** [comment]
trigger pmd

**Comment by ivan-83** [comment]
trigger messagecheck

**Comment by ivan-83** [comment]
trigger codeformat

**Comment by ivan-83** [comment]
trigger uts

**Comment by ivan-83** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by ivan-83** [comment]
trigger review

**Review by yogesh-chauhan — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 4 files at r2, 2 of 4 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52975)* status: Some of the checks are too old. 6 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by acasian-harness — APPROVED** [review]

**Comment by ivan-83** [comment]
trigger review

**Pull request merged**